### DISCUSSION TITLE
Is there no dark theme in Qucs-S 24.2 on Windows?

### DISCUSSION THREAD

**Discussion by deralbert** [ra3xdh/qucs_s#683] [Q&A] [answered]
I am using Qucs-S 24.2 and want to use the application completely in dark mode. I have set two properties called "Panel icons theme" and "Components icon theme" to "dark" and restarted the application. However, nothing has changed. Could this be a bug?

**Comment by ra3xdh** [comment]
It is not a bug. You cannot set the dark theme from application. The Qt application always uses the system theme. These settings were needed to set the inverse icons colors for the toolbars if the system wide theme was not recognized correctly. See #66 and #69. These settings are not needed anymore after switching to SVG icons. I forgot to remove it from the settings dialog.

**Comment by deralbert** [comment]
My system theme is dark but the application doesn't recognize it for some reason. I use Windows 10.

**Comment by ra3xdh** [comment]
This may be a restriction of the Qt toolkit for Windows. Qt applications for Linux platform use system theme without problem. There is no mean to set the theme from inside Qt application.

**Reply by deralbert** [reply]
Could you please point me to the place in the code (or at least the file/files) where the theme is set when the program starts? I've only looked superficially and didn't find it right away. If I have time, I'll try to fix this behavior or at least find the cause of it.

**Comment by ra3xdh** [comment]
> Could you please point me to the place in the code (or at least the file/files) where the theme is set

There is no such place. It is not possible to change the theme in Qt application manually by means of Qt. The Qt application can only take the system theme.

**Reply by deralbert** [reply, marked as answer]
I've looked superficially, and it does seem like you can activate a desired theme programmatically, but perhaps I'm misunderstanding something. I haven't worked with Qt a lot.

I managed to start the application in a semi-dark mode (for Qt 5): `.\qucs-s.exe -platform windows:darkmode=2`
Just to note: For Qt 6.5, it would be `-style fusion`.

<img width="640" alt="Screenshot_137" src="https://github.com/ra3xdh/qucs_s/assets/909718/68744cbe-22fb-4b9d-9b50-7408be0e6cd6">
